lale/lib/aif360/util.py

```diff
@@ -636,11 +638,13 @@ def __init__(
 
     def __call__(self, estimator, X, y):
         disp_impact = self.disparate_impact_scorer(estimator, X, y)
-        if np.isnan(disp_impact):  # empty privileged or unprivileged groups
-            return np.NAN
         r2 = self.r2_scorer(estimator, X, y)
+        if np.isnan(disp_impact):  # empty privileged or unprivileged groups
+            return r2
         assert r2 <= 1.0 and 0.0 <= disp_impact, (r2, disp_impact)
-        if disp_impact <= 1.0:
+        if disp_impact == 0.0:
+            return 0.0
```

Comment:
I think in this case, instead of returning 0.0, the result should be something like `-sys.float_info.max`. The worst possible disparate impact should also yield the worst possible combined metric.

Reply:
Thanks, please check the commits below.